pkg/controllers/deprovisioning/reporter.go

```diff
@@ -0,0 +1,41 @@
+package deprovisioning
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/logging"
+
+	deprovisioningevents "github.com/aws/karpenter-core/pkg/controllers/deprovisioning/events"
+	"github.com/aws/karpenter-core/pkg/events"
+	"github.com/aws/karpenter-core/pkg/utils/pretty"
+)
+
+// Reporter is used to periodically report node statuses regarding deprovisioning. This gives observers awareness of why
+// deprovisioning of a particular node isn't occurring.
+type Reporter struct {
+	cm       *pretty.ChangeMonitor
+	recorder events.Recorder
+}
+
+func NewReporter(recorder events.Recorder) *Reporter {
```

Comment:
I pulled this out thinking it might be useful for reporting upgrade and drift as well.